test/vm-with-real-chain-1st-1000.spec.js

```diff
@@ -0,0 +1,69 @@
+'use strict'
+/* eslint-env mocha */
+
+const ethereum = require('../src')
+const expect = require('chai').expect
+const Block = require('ethereumjs-block')
+// const BlockHeader = require('ethereumjs-block/header.js')
+// const parallel = require('run-parallel')
+const async = require('async')
+// const utils = require('ethereumjs-util')
+// const rlp = utils.rlp
+
+const thousand = require('./data/real-chain/first-1000-blocks.json')
+const states = require('./data/example-chain/states.json')
+
+describe.only('process 1st 1000 blocks', () => {
+  let eth
+
+  it('spawn a node', () => {
+    eth = new ethereum.Node()
+  })
+
+  it('load pre data to setup vm', (done) => {
+    eth.vm.setup(states.pre, done)
+  })
+
+  it('add genesis block', (done) => {
+    const genesis = new Block(new Buffer(thousand[0].slice(2), 'hex'))
+    eth.vm._blockchain.putGenesis(genesis, done)
+  })
+
+  it.skip('add more blocks', (done) => {
+    async.eachSeries(thousand.slice(1), eachBlock, done)
+
+    function eachBlock (raw, cb) {
+      let block
+      try {
+        block = new Block(new Buffer(raw.slice(2), 'hex'))
+
+        // forces the block into thinking they are homestead
+        block.header.isHomestead = () => { return true }
+        block.uncleHeaders.forEach((uncle) => {
+          uncle.isHomestead = () => { return true }
+        })
+
+        console.log('->', block.difficulty)
+
+        eth.vm._blockchain.putBlock(block, (err) => {
+          expect(err).to.not.exist
+          cb
+        })
+      } catch (err) { cb(err) }
+    }
+  })
```

Comment:
@wanderer this is the test that is failing, it tries to process the first thousand blocks of the real ethereum chain, but fails on the first one with:

```
  1) process 1st 1000 blocks add more blocks:
     Uncaught AssertionError: expected 'invalid Difficulty' to not exist
      at eth.vm._blockchain.putBlock (test/vm-with-real-chain-1st-1000.spec.js:49:29)
      at node_modules/ethereumjs-blockchain/index.js:114:5
      at node_modules/ethereumjs-blockchain/node_modules/async/lib/async.js:726:13
```

To try it out, git clone, npm i, take of the `skip` from this test and npm run test. Note that putting the genesis block runs fine though :)